src/popup/components/Modals/TransferReceive.vue

```diff
@@ -101,95 +115,102 @@ export default {
     Scrollable,
     AddressFormatted,
     ShareIcon,
+    CopyText,
   },
-  mixins: [CopyMixin],
   props: {
     defaultAmount: { type: [String, Number], default: null },
     tokenContractId: { type: [String, Number], default: null },
   },
-  subscriptions() {
-    return pick(this.$store.state.observables, [
-      'balance',
-      'balanceCurrency',
-    ]);
-  },
-  data() {
-    return {
-      IS_MOBILE_DEVICE,
-      amount: null,
-      selectedAsset: null,
-    };
-  },
-  computed: {
-    ...mapGetters('fungibleTokens', [
-      'getAeternityToken',
-    ]),
-    ...mapGetters([
-      'account',
-      'activeNetwork',
-    ]),
-    ...mapState('fungibleTokens', [
-      'availableTokens',
-    ]),
-    computedAddress() {
-      return (this.amount > 0)
-        ? `${this.account.address}?${this.getTokenInfo(true).substring(1)}`
-        : this.account.address;
-    },
-    explorerUrl() {
-      return `${this.activeNetwork.explorerUrl}/account/transactions/${this.account.address}`;
-    },
-  },
-  created() {
-    this.amount = this.defaultAmount;
-    if (this.tokenContractId && this.availableTokens[this.tokenContractId]) {
-      this.selectedAsset = this.availableTokens[this.tokenContractId];
-    } else {
-      this.selectedAsset = this.getAeternityToken({
-        tokenBalance: this.balance,
-        balanceCurrency: this.balanceCurrency,
-      });
-    }
-  },
-  methods: {
-    getTokenInfo(link = false) {
-      if (this.amount <= 0) return '';
-      const token = this.selectedAsset && (this.selectedAsset.symbol === 'AE'
+  setup(props, { root }) {
+    const amount = ref<string | number>(0);
```

Comment:
const amount = ref<number>(0);

Reply:
Why? The amount can be a string also. For example if user use the period instead of dot.

Reply:
maybe we need refactor input?

Reply:
Using number option to v-model for this. Resolving.